docs/en/latest/deployment-modes.md

```diff
@@ -115,73 +115,160 @@ This method is more suitable for two types of users:
+    ```json
+    {
+      "routes_conf_version": 12,
+      "upstreams_conf_version": 102,
+      "routes": [],
+      "upstreams": []
+    }
     ```
 
-    The update API validates the input and returns an error if it is invalid. If the configuration is accepted, it responds with a `202 Accepted` status and includes the latest configuration version in the `X-APISIX-Conf-Version` header.
+    APISIX compares each provided `<resource>_conf_version` against its in-memory `<resource>_conf_version` for that resource type:
+
+  - **Greater than** the current `conf_version`
+    If your `<resource>_conf_version` is **higher**, APISIX will **rebuild/reset** that resource type’s data to match your payload.
+
+  - **Equal to** the current `conf_version`
+    If it is **equal**, APISIX treats the resource as **unchanged** and **ignores** it (no data is rebuilt).
+
+  - **Less than** the current `conf_version`
+    If it is **lower**, APISIX considers your update **stale** and **rejects** the request for that resource type with a **400 Bad Request**.
+
+* modifiedIndex per resource
+
+    Allow setting an index for each resource, APISIX compares it to its modifiedIndex to determine whether to accept the update.
+
+**Example:**
+
+1. get configuration
+
+```shell
+curl -X GET http://127.0.0.1:9180/apisix/admin/configs \
+    -H "X-API-KEY: <apikey>" \
+    -H "Accept: application/json" ## or application/yaml
+```
+
+This returns the current configuration in JSON or YAML format.
+
+```json
+{
+    "consumer_groups_conf_version": 1000,
+    "consumers_conf_version": 1000,
+    "global_rules_conf_version": 1000,
+    "plugin_configs_conf_version": 1000,
+    "plugin_metadata_conf_version": 1000,
+    "protos_conf_version": 1000,
+    "routes_conf_version": 1000,
+    "secrets_conf_version": 1000,
+    "services_conf_version": 1000,
+    "ssls_conf_version": 1000,
+    "upstreams_conf_version": 1000
+}
+```
+
+2. full update
+
+```shell
+curl -X PUT http://127.0.0.1:9180/apisix/admin/configs \
+    -H "X-API-KEY: <apikey>" \
+    -H "Content-Type: application/json" ## or application/yaml \
+    -d '{}'
+```
+
+3. update based on resource type
+
+In APISIX memory, the current configuration is:
+
+```json
+{
+    "routes_conf_version": 1000,
+    "upstreams_conf_version": 1000,
+}
+
+Due `upstreams_conf_version: 1001` > `upstreams_conf_version: 1000`, APISIX will update the upstreams configuration:
+
+```shell
+curl -X PUT http://127.0.0.1:9180/apisix/admin/configs \
+  -H "X-API-KEY: ${API_KEY}" \
+  -H "Content-Type: application/json" \
+  -d '
+{
+    "routes_conf_version": 1000,
+    "upstreams_conf_version": 1001,
+    "routes": [
+        {
+            "id": "r1",
+            "uri": "/hello",
+            "upstream_id": "u1"
+        }
+    ],
+    "upstreams": [
+        {
+            "id": "u1",
+            "nodes": {
+                "127.0.0.1:1980": 1,
+                "127.0.0.1:1980": 1
+            },
+            "type": "roundrobin"
```

Comment:
add a `modifiedIndex` field to those resources.

Reply:
fixed.